language/golang/build-images.md

```diff
@@ -0,0 +1,388 @@
+
+```dockerfile
+COPY go.mod .
+COPY go.sum .
+```
+
+Now that we have the module files inside the Docker image that we are building, we can use the `RUN` command to execute the command `go mod download` there as well. This works exactly the same as if we were running `go` locally on our machine, but this time these Go modules will be installed into the a directory inside our image.
+
+```dockerfile
+RUN go mod download
+```
+
+At this point, we have an image that is based on Go environment version 1.16 (or a later minor version, since we had specified `1.16` as our tag in the `FROM` command) and we have installed our dependencies. 
+
+The next thing we need to do is to add our source code into the image. We’ll use the `COPY` command just like we did with our module files before.
+
+```dockerfile
+COPY *.go .
+```
+
+This `COPY` command uses a wildcard to copy all files with `.go` extension located in the current directory into the image. 
+
+Now, we would like to compile our application. To that end, we use the familiar `RUN` command.
+
+```dockerfile
+RUN go build -o /docker-gs-ping
+```
+
+This should be familiar. The result of that command will be a static application binary named `docker-gs-ping` and located in the root of the filesystem of the image that we are building. We could have put the binary into any other place we desire inside that image, there is no special significance for placing it in the root.
+
+Now, all that is left to do is to tell Docker what command to execute when our image is used to start a container. 
+
+We do this with the `CMD` command.
+
+```dockerfile
+CMD [ "/docker-gs-ping" ]
+```
+
+Here's the complete `Dockerfile`. The file in the repository that you cloned may also contain comments. They always begin with a `#` symbol and make no difference to Docker. The comments are there for the convenience of humans tasked to maintain the `Dockerfile`.
+
+```dockerfile
+FROM golang:1.16-alpine
+
+WORKDIR /app
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+COPY *.go .
+
+RUN go build -o /docker-gs-ping
+
+EXPOSE 8080
+
+CMD [ "/docker-gs-ping" ]
+```
+
+You can also find this `Dockerfile` for this part in [go-docker](https://github.com/olliefr/go-docker) repo.
+
+## Build the image
+
+Now that we've created our `Dockerfile`, let’s build an image from it. The `docker build` command creates Docker images from a Dockerfile and a "context". A build's _context_ is the set of files located in the specified `PATH` or `URL`. The Docker build process can access any of the files located in the context.
+
+The build command optionally takes a `--tag` flag. This flag is used to label the image with a string value, which is easy for humans to read and recognise. If you do not pass a `--tag`, Docker will use `latest` as the default value.
+
+Let's build our first Docker image!
+
+```shell
+$ docker build --tag docker-gs-ping .
```

Comment:
Should we keep `go-docker` as local image name? It is used later to update the image.

Reply:
`go-docker` was a different application, it's been retired since 😂 I'm sticking to the ping example for the first part and CI/CD. And I'll have a different example for the "Develop" part since it involves the database.